src/meta/src/hummock/hummock_manager.rs

```diff
@@ -595,21 +591,16 @@ where
                         .flat_map(|v| v.snapshot_id.clone())
                         .fold(max_committed_epoch, std::cmp::min)
                 };
+                commit_multi_var!(
+                    self,
+                    Some(assignee_context_id),
+                    compact_status,
+                    compact_task_assignment
+                )?;
                 Ok(Some(compact_task))
             }
         };
 
-        if should_commit {
-            commit_multi_var!(
-                self,
-                Some(assignee_context_id),
-                compact_status,
-                compact_task_assignment
-            )?;
-        } else {
-            abort_multi_var!(compact_status);
```

Comment:
@wenym1 Is it safe to not call abort here?

Reply:
Seems that our updated rust-toolchain has no longer required us to drop variables on every branch.